### PR TITLE
Change from namespace se3 to pinocchio

### PR DESCRIPTION
**Pull request by jcarpent**
Related to discussions in #570.

The former se3 is still valid. The deprecated message is not shown if `PINOCCHIO_ENABLE_COMPATIBILITY_WITH_VERSION_1` is defined.

### CODE REVIEW

File: src/deprecated-namespaces.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// Pinocchio If not, see
+// <http://www.gnu.org/licenses/>.
+
+#ifndef __pinocchio_deprecated_namespaces_hpp__
+#define __pinocchio_deprecated_namespaces_hpp__
+
+#if PINOCCHIO_ENABLE_COMPATIBILITY_WITH_VERSION_1 // do not warn
+  #define se3 ::pinocchio
+#else
+  #define se3 PINOCCHIO_PRAGMA_MESSAGE_CALL("The se3 namespace has been deprecated since version 2.0.0. Please use namespace pinocchio instead.") ::pinocchio
```
Comment (jmirabel) [inline src/deprecated-namespaces.hpp]: I think this is a very bad idea. Using a macro does not make the code compatible with Pinocchio 1.

I would merely do:
```cpp
#if PINOCCHIO_ENABLE_COMPATIBILITY_WITH_VERSION_1 // do not warn
  namespace se3 = ::pinocchio;
#endif
```

File: unittest/liegroups.cpp [hunk 2]
```diff
@@ -544,13 +544,13 @@ BOOST_AUTO_TEST_CASE ( test_size )
   BOOST_CHECK (se2.name () == "SE(2)");
   BOOST_CHECK (se2.neutral () == neutral);
   // SE(3): neutral = [0, 0, 0, 0, 0, 0, 1]
-  SpecialEuclideanOperationTpl <3,double> se3;
+  SpecialEuclideanOperationTpl <3,double> se3_group;
   neutral.resize (7); neutral.setZero ();
```
Comment (jmirabel) [inline unittest/liegroups.cpp]: To emphasize the reason why the `#define se3 pinocchio` is bad: you had to modify a unit test, which can be considered as user code.

This is not what I call compatible. I think you should remove this change.

File: src/deprecated-namespaces.hpp (hunk 1 above)
Comment (jcarpent) [reply src/deprecated-namespaces.hpp]: I understand. This also means that users would be only informed by reading release notes.

File: unittest/liegroups.cpp (hunk 2 above)
Comment (jcarpent) [reply unittest/liegroups.cpp]: I have the same opinion. Let's go for that then.

**Comment by jcarpent** [comment]
@jmirabel I have made the change. Can you approve?

**Comment by jmirabel** [comment]
Thank you.

**Comment by jcarpent** [comment]
Thanks to you too.

**Pull request merged**